src/praisonai-train/praisonai_train/train/llm/trainer.py

```
@@ -213,9 +242,65 @@ def validate_config(self):
         if self._flag(self.config.get("ollama_save")) and not self.config.get("ollama_model"):
             raise ValueError("ollama_model is required when ollama_save is enabled.")
-        for key in self.config:
-            if key not in self.KNOWN_KEYS:
-                print(f"WARNING: ignoring unknown config key '{key}' (typo, or not supported).")
+
+        # --- Preflight: fail FAST with friendly messages BEFORE any model/GPU load,
+        # so a non-developer gets a clear, actionable error in seconds instead of a
+        # traceback after minutes of downloads. ---
+
+        # GPU: fine-tuning cannot run on CPU. Catch it before Unsloth tries.
+        if not torch.cuda.is_available():
```

Ruff (0.15.21)

[error] Undefined name `torch`

(F821)